src/sys/ptrace.rs

```diff
@@ -1,12 +1,23 @@
 use std::{mem, ptr};
 use {Errno, Error, Result};
-use libc::{c_void, c_long, siginfo_t};
+use libc::{self, c_void, c_long, siginfo_t};
 use ::unistd::Pid;
 
 pub mod ptrace {
     use libc::c_int;
 
-    pub type PtraceRequest = c_int;
+    /// The datatype used for the `request` argument to `ptrace`
+    #[cfg(any(all(target_os = "linux", arch = "s390x"),
+              all(target_os = "linux", target_env = "gnu")))]
+    #[doc(hidden)]
+    pub type ptrace_request_type = ::libc::c_uint;
+
+    #[cfg(not(any(all(target_os = "linux", arch = "s390x"),
+                  all(target_os = "linux", target_env = "gnu"))))]
+    #[doc(hidden)]
+    pub type ptrace_request_type = ::libc::c_int;
```

Comment:
Enums are `c_int` by default in C. But I'm following the argument type dictated by `libc`'s definition of `ptrace()` across all supported platforms.